sky/clouds/service_catalog/aws_catalog.py

```diff
@@ -35,26 +37,52 @@
                                 pull_frequency_hours=_PULL_FREQUENCY_HOURS)
 
 
-def _apply_az_mapping(df: 'pd.DataFrame') -> 'pd.DataFrame':
+def _apply_az_mapping(df: pd.DataFrame) -> pd.DataFrame:
     """Maps zone IDs (use1-az1) to zone names (us-east-1x).
 
-    Such mappings are account-specific and determined by AWS.
+    Such mappings are account-specific and determined by AWS. We fetch the
+    mappings from AWS, which requires AWS credentials. If the user does not
+    have AWS credentials configured, we use zone id as the zone name, which
+    is only used for show-gpus. It is ok to use the default mapping because
+    the user will not be able to provision instances with those wrong
+    availablity zones due to the lack of credentials.
+
+    The mappings will also serve to remove from '_df' the regions that are
+    not supported by the user account.
 
     Returns:
         A dataframe with column 'AvailabilityZone' that's correctly replaced
         with the zone name (e.g. us-east-1a).
     """
-    az_mapping_path = common.get_catalog_path('aws/az_mappings.csv')
+    aws_enabled, _ = aws.AWS().check_credentials()
+    if aws_enabled:
+        user_identity = aws.AWS().get_current_user_identity()
```

Comment:
This seems visibly slower on my laptop: 0.9s -> 2.1s for `time sky show-gpus --cloud aws`.

Reply:
Just optimized it for `show-gpus`. Please feel free to have another try. Thanks!